File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        pref = ""
        strs.sort()
        a = strs[0]
        b = strs[-1]
        N = len(a) if len(a) < len(b) else len(b)
        for i in range(N):
            if a[i] == b[i]:
                pref += a[i]
            else:
                return pref
        return pref

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
import java.util.Arrays;

class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String a = strs[0];
        String b = strs[strs.length-1];
        String prefix = "";
        int i = 0;
        while(i < a.length() && i < b.length()){
            if(a.charAt(i) != b.charAt(i)){
                break;
            }
            i++;
        }
        prefix = a.substring(0, i);
        return prefix;
    }
}